# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/agnostic/bytestreams/AnnexB.h
# Commit: 412f97da1656
# Full Hash: 412f97da16567da3a0c2a6adc6dbde27a28dea22
# Author: alwu <alwu@mozilla.com>
# Date: 2023-10-11 09:22:03
# Regressor Bug: 1852333
# File Overlap Count: 4
# Description:
#   Bug 1852333 - part2 : detect config change for HEVC stream. r=media-playback-reviewers,padenot
#   
#   This patch detects inband change for the HEVC playback so that we can
#   recreate decoder if the new config is different from the previous one.
#   
# ==============================================================================

diff -r b15b2e7c785a -r 412f97da1656 dom/media/platforms/agnostic/bytestreams/AnnexB.h
--- a/dom/media/platforms/agnostic/bytestreams/AnnexB.h	Wed Oct 11 00:57:39 2023 +0000
+++ b/dom/media/platforms/agnostic/bytestreams/AnnexB.h	Wed Oct 11 00:57:40 2023 +0000
@@ -43,6 +43,8 @@
   static bool ConvertSampleToAVCC(
       mozilla::MediaRawData* aSample,
       const RefPtr<mozilla::MediaByteBuffer>& aAVCCHeader = nullptr);
+  // Convert a sample from Annex B to HVCC. An HVCC extradata must not be set.
+  static bool ConvertSampleToHVCC(mozilla::MediaRawData* aSample);
 
   // Covert sample to 4 bytes NALU byte stream.
   static mozilla::Result<mozilla::Ok, nsresult> ConvertAVCCTo4BytesAVCC(